extensions-core/s3-extensions/src/main/java/org/apache/druid/storage/s3/output/S3StorageConnector.java

```diff
@@ -280,8 +280,11 @@ public OutputStream write(String path) throws IOException
   public void deleteFile(String path) throws IOException
   {
     try {
+      final String fullPath = objectPath(path);
+      log.debug("Deleting file at bucket: %s, path: %s", config.getBucket(), fullPath);
+
       S3Utils.retryS3Operation(() -> {
-        s3Client.deleteObject(config.getBucket(), objectPath(path));
+        s3Client.deleteObject(config.getBucket(), fullPath);
```

Comment:
Is there a reason we don't use deleteObjects for deleting a single file as well? I wonder if refactoring it to also pass through S3Utils like deleteFiles would be cleaner.

Reply:
Since its a single file, I think its okay to have s3Client.deleteObject() for it.